index.js

```diff
@@ -3,156 +3,131 @@
-        formatted = format(message, a, b)
-      } else if (a) {
-        formatted = format(message, a)
-      } else {
-        formatted = message
-      }
+/**
+ * Creates a warning item.
+ * @function
+ * @memberof processWarning
+ * @param {string} name - The name of the warning.
+ * @param {string} code - The code associated with the warning.
+ * @param {string} message - The warning message.
+ * @param {ProcessWarningOptions} [opts={}] - Options for creating the warning.
+ * @returns {WarningItem} The created warning item.
+ * @throws {Error} Throws an error if name, code, or message is empty, or if opts.unlimited is not a boolean.
+ */
+function createWarning (name, code, message, { unlimited = false } = {}) {
+  if (!name) throw new Error('Warning name must not be empty')
+  if (!code) throw new Error('Warning code must not be empty')
+  if (!message) throw new Error('Warning message must not be empty')
+  if (typeof unlimited !== 'boolean') throw new Error('Warning opts.unlimited must be a boolean')
 
-      return {
-        code,
-        name,
-        message: formatted
-      }
-    }
+  code = code.toUpperCase()
 
-    Object.assign(opts, { unlimited })
-    emitted.set(code, unlimited)
-    codes[code] = buildWarnOpts
+  return new WarningItem(name, code, message, unlimited)
+}
 
-    return codes[code]
+/**
+ * Represents a warning item with details.
+ * @class
+ * @memberof processWarning
+ * @param {string} name - The name of the warning.
+ * @param {string} code - The code associated with the warning.
+ * @param {string} message - The warning message.
+ * @param {boolean} unlimited - If true, allows unlimited emissions of the warning.
+ */
+class WarningItem {
+  constructor (name, code, message, unlimited) {
+    this.name = name
+    this.code = code
+    this.message = message
+    this.unlimited = unlimited
+    this.emitted = false
   }
 
   /**
-   * A wrapper for {@link ProcessWarningManager#create} that builds a new
-   * deprecation warning. This method is equivalent to passing
-   * `name = 'DeprecationWarning'` to the create method.
-   *
-   * Deprecation warnings have extended support for the Node.js CLI options:
-   * `--throw-deprecation`, `--no-deprecation`, and `--trace-deprecation`.
-   *
-   * @memberof! ProcessWarningManager
-   * @instance
-   *
-   * @param {string} code
-   * @param {string} message
-   * @param {CreateOptions} opts
-   * @returns {ProcessWarningBuilder}
+   * Emits the warning.
+   * @param {*} [a] Possible message interpolation value.
+   * @param {*} [b] Possible message interpolation value.
+   * @param {*} [c] Possible message interpolation value.
    */
-  function createDeprecation (code, message, opts = {}) {
-    return create('DeprecationWarning', code, message, opts)
+  emit (a, b, c) {
+    if (this.emitted === true && this.unlimited === false) {
+      return
+    }
+    this.emitted = true
+
+    process.emitWarning(this.format(a, b, c), this.name, this.code)
```

Comment:
```suggestion
    if (this.unlimited === false) {
      this.emit = () => {}
    }

    process.emitWarning(this.format(a, b, c), this.name, this.code)

    this.emitted = true
```
When I do this, then the benchmarks hits the roof.